### PR TITLE
[ts-sdk-v2] Add add faucet client and fundaccount query

### PR DESCRIPTION
**Pull request by heliuchuan**
### Description
Adds the faucet client

### CODE REVIEW

File: ecosystem/typescript/sdk_v2/src/api/faucet.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { fundAccount } from "../internal/faucet";
```
Comment (banool) [inline ecosystem/typescript/sdk_v2/src/api/faucet.ts:1]: We should add support for using auth tokens. Rust client for reference: https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos-rest-client/src/faucet.rs#L53.

**Review by banool — CHANGES_REQUESTED** [review]

File: ecosystem/typescript/sdk_v2/src/api/faucet.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    /**
+     * This creates an account if it does not exist and mints the specified amount of
+     * coins into that account
+     * 
+     * @param address Hex-encoded 16 bytes Aptos account address wich mints tokens
+     * @param amount Amount of tokens to mint
+     * @param timeoutSecs Timeout in seconds. Defaults to 20 seconds.
+     * @returns Hashes of submitted transactions
+     */
+    async fundAccount(args: { accountAddress: HexInput, amount: number, timeoutSecs?: number }): Promise<string[]> {
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/api/faucet.ts]: for consistency
```suggestion
    async fundAccount(args: { accountAddress: HexInput, amount: number, timeoutSecs?: number }): Promise<Array<string>> {
```

**Review by 0xmaayan — APPROVED** [review]
LGTM.
This PR blocked by https://github.com/aptos-labs/aptos-core/pull/10374/files, right?

File: ecosystem/typescript/sdk_v2/src/api/faucet.ts (hunk 1 above)
Comment (heliuchuan) [reply ecosystem/typescript/sdk_v2/src/api/faucet.ts:1]: I believe it's already supported via the aptosConfig through the constructor?  is there a need to add a withAuthToken method?

Comment (0xmaayan) [reply ecosystem/typescript/sdk_v2/src/api/faucet.ts:1]: Yes thats true. Can set a Token on AptosConfig. That token would actually go with all requests (fullnode, indexer, faucet) though - I wonder if that could cause issues. @heliuchuan might worth to test that out

**Pull request merged**